apps/aecore/test/aec_blocks_tests.erl

```diff
@@ -71,23 +71,27 @@ network_serialization_test_() ->
                           ?TEST_MODULE:deserialize_from_network(SerializedBlock2))
      end}].
 
+%% Each block has at most one coinbase transaction
+%% If a coinbase transaction in a block has a malformed hash, then the block does not validate
+%% 
 validate_test_() ->
     [fun() ->
-             SignedCoinbase = #signed_tx{data = #coinbase_tx{}},
+             SignedCoinbase = aec_tx_sign:sign(#coinbase_tx{}, <<"key">>),
              Block = #block{txs = [SignedCoinbase, SignedCoinbase]},
              ?assertEqual({error, multiple_coinbase_txs}, ?TEST_MODULE:validate(Block))
      end,
      fun() ->
-             SignedCoinbase = #signed_tx{data = #coinbase_tx{}},
+             SignedCoinbase = aec_tx_sign:sign(#coinbase_tx{}, <<"key">>),
              CorrectTxs = [SignedCoinbase],
-             MalformedTxs = [SignedCoinbase, #signed_tx{data = #coinbase_tx{account = <<"malformed_account">>}}],
+             MalformedTxs = [SignedCoinbase, aec_tx_sign:sign(#coinbase_tx{account = <<"malformed_account">>}, <<"key">>)],
              {ok, MalformedTree} = aec_txs_trees:new(MalformedTxs),
              {ok, MalformedRootHash} = aec_txs_trees:root_hash(MalformedTree),
              Block = #block{txs = CorrectTxs, txs_hash = MalformedRootHash},
              ?assertEqual({error, malformed_txs_hash}, ?TEST_MODULE:validate(Block))
      end,
      fun() ->
-             SignedCoinbase = #signed_tx{data = #coinbase_tx{}},
+             SignedCoinbase = aec_tx_sign:sign(#coinbase_tx{}, <<"key">>),
+             %% This is weird, we test that a coinbase that is signed with a key that is not the account key does validate!!
```

Comment:
Aren't we testing that nothing bad happens if the coinbase tx is bad? The tree will not contain the coinbase tx in this case I think!?